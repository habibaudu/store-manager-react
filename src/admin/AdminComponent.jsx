import React from 'react';
import { Link } from 'react-router-dom';
import img2 from '../img/download.jpeg';
class AdminComponent extends React.Component {
  componentDidMount() {
    this.props.fetchAllProducts();
  }

  render() {
    return (
      <div>
        <nav>
          <div className="logo">
            <h4>Store Manager</h4>
          </div>

          <ul className="nav-links">
            <li>
              {' '}
              <div className="dropdown">
                <span className="dropbtn">User</span>
                <div className="dropdown-content">
                  <Link to="/createAttendant">Create Attendant</Link>
                </div>
              </div>
            </li>
            <li>
              <img src={img2} alt="adminlogo" id="adminlogo" />
            </li>
          </ul>
          <div className="burger">
            <div className="line1" />
            <div className="line2" />
            <div className="line3" />
          </div>
        </nav>

        <div className="adminbox">
          <Link to="/addproduct">
            <button className="addproduct" id="butt">
              +Add Product
            </button>
          </Link>
          <div>
            {this.props.payload ? (
              <table style={{ overflowX: 'auto' }}>
                <thead>
                  <tr>
                    <th>ProductID</th>
                    <th>Name</th>
                    <th>Quantity</th>
                    <th>Price</th>
                    <th>Minimum Allowed</th>
                    <th>Description</th>
                    <th>Modified At</th>
                    <th>Delete</th>
                  </tr>
                </thead>
                <tbody id="allproducts">
                  {this.props.payload.map(product => {
                    const {
                      id,
                      productname,
                      price,
                      minimum,
                      quantity,
                      description,
                      modified_date,
                    } = product;
                    return (
                      <tr key={id}>
                        <td>{id}</td>
                        <td>{productname}</td>
                        <td>{quantity}</td>
                        <td>{price}</td>
                        <td>{minimum}</td>
                        <td>{description}</td>
                        <td>{modified_date}</td>
                        <td>
                          <button
                            className="bt3"
                            onClick={() => this.props.deleteProduct(id)}
                          >
                            delete
                          </button>
                        </td>
                      </tr>
                    );
                  })}
                </tbody>
              </table>
            ) : (
              <h3>loading product please wait...</h3>
            )}
          </div>
        </div>
      </div>
    );
  }
}

export default AdminComponent;
