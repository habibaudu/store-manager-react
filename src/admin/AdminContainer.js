import { connect } from 'react-redux';
import { fetchAllProducts } from './duck';
import AdminComponent  from './AdminComponent';



const mapStateToProps = (state) => {
  return {
          payload :state.adminReducer.payload,
          
        }
};

const mapDispatchToProps = dispatch => {
  return {
    fetchAllProducts: () =>
      dispatch(fetchAllProducts()),
  };
};
const AdminContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(AdminComponent);

export { AdminContainer, mapDispatchToProps, mapStateToProps };
