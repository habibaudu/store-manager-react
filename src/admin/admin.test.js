import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Redirect, MemoryRouter } from 'react-router-dom';
import axios from 'axios';
import AdminComponent from './AdminComponent';
import { AdminContainer, mapDispatchToProps } from './AdminContainer';
import {
  actions,
  types,
  constants,
  adminReducer,
  deleteProduct,
  fetchAllProducts,
  addProduct,
} from './duck';

jest.mock('axios');

const { setAllProducts, addProductStatus } = actions;
const props = {
  fetchAllProducts: jest.fn(),
  deleteProduct: jest.fn(),
};

describe('ADMIN TEST SUITE', () => {
  describe('Admin Component', () => {
    it('should render the Admin Page', () => {
      const component = shallow(<AdminComponent {...props} />);
      expect(component.exists()).toBe(true);
      expect(component).toMatchSnapshot();
    });
    it('it should submit the form ', () => {
      const props = {
        deleteProduct,
        fetchAllProducts,
        payload: [
          {
            productname: 'shoe lace',
            quantity: 3,
            minimum: 4,
            description: 'a good product',
            price: 230,
          },
          {
            productname: 'shoe lace2',
            quantity: 33,
            minimum: 4,
            description: 'a good product',
            price: 230,
          },
        ],
      };
      const payload = [
        {
          productname: 'shoe lace',
          quantity: 3,
          minimum: 4,
          description: 'a good product',
          price: 230,
        },
        {
          productname: 'shoe lace2',
          quantity: 33,
          minimum: 4,
          description: 'a good product',
          price: 230,
        },
      ];
      const component = shallow(<AdminComponent {...props} />);
    });

    it('should render the Admin container', () => {
      const component = shallow(<AdminContainer />);
      expect(component.exists()).toBe(true);
      expect(component).toMatchSnapshot();
    });
  });

  it('should ensures mapDispatchToProps dispatches the specified actions', () => {
    const mock = configureStore([thunk]);
    const store = mock(props);
    const component = shallow(
      <Provider store={store}>
        <AdminContainer />
      </Provider>,
    );
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).fetchAllProducts();
    mapDispatchToProps(dispatch).deleteProduct();
  });
});

describe('TEST FOR GET PRODUCT', () => {
  const payload = {
    data: {
      productname: 'productname',
      id: 1,
      quantity: 1,
      minimum: 2,
    },
  };
  const mockStore = configureStore([thunk]);
  const store = mockStore({});
  let wrapper;
  const response = { message: 'fetchall product' };
  axios.get.mockImplementation(() => Promise.resolve(response));
  store.dispatch(fetchAllProducts()).then(() => {
    store.dispatch(setAllProducts(payload));
  });
});

describe('TEST FOR DELETE PRODUCT', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({});
  let wrapper;
  const response = { message: 'delete product' };
  axios.delete.mockResolvedValue(response);
  store.dispatch(deleteProduct()).then(() => {});
});

describe('TEST FOR ADD PRODUCT', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({});
  let wrapper;
  const response = { message: 'add product' };

  axios.post.mockImplementation(() => Promise.resolve(response));

  store.dispatch(addProduct()).then(() => {
    store.dispatch(addProductStatus(constants.PRODUCT_SUCESSFULLY_ADDED));
  });

  it('Should test for product add: failure', () => {
    const product = {};
    const store = mockStore({});
    const response = { data: 'product add not sucessfull' };
    axios.patch.mockImplementation(() => Promise.reject(response));

    store.dispatch(addProduct(product)).catch(() => {
      store.dispatch(addProductStatus(constants.ADD_ERROR));
    });
  });
});
