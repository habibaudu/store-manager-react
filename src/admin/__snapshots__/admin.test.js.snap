// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADMIN TEST SUITE Admin Component should render the Admin Page 1`] = `
<div>
  <nav>
    <div
      className="logo"
    >
      <h4>
        Store Manager
      </h4>
    </div>
    <ul
      className="nav-links"
    >
      <li>
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/createAttendant"
        >
          Create Attendant
        </Link>
      </li>
      <li>
        <img
          alt="adminlogo"
          id="adminlogo"
          src={Object {}}
        />
      </li>
    </ul>
    <div
      className="burger"
    >
      <div
        className="line1"
      />
      <div
        className="line2"
      />
      <div
        className="line3"
      />
    </div>
  </nav>
  <div
    className="adminbox"
  >
    <Link
      replace={false}
      to="/addproduct"
    >
      <button
        className="addproduct"
        id="butt"
      >
        +Add Product
      </button>
    </Link>
    <div
      style={
        Object {
          "overflowX": "auto",
        }
      }
    >
      <h3>
        loading product please wait...
      </h3>
    </div>
  </div>
</div>
`;

exports[`ADMIN TEST SUITE Admin Component should render the Admin container 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
